### DISCUSSION TITLE
Proposing a New Validation Rule for Longitude, Latitude, and Street View Camera Angle

### DISCUSSION THREAD

**Discussion by asfiaaiman** [laravel/framework#52804] [Ideas]
I propose adding a new custom validation rule to Laravel that will handle validation for longitude, latitude, and Street View camera angle values. This rule will also account for optional degree signs (°) that might be included with these values.

1. Longitude Validation:

Criteria: Should be within -180 to 180 degrees.
Example Valid Inputs: -73.935242, 45.0, 180°
Example Invalid Inputs: -181, 200, abc

2. Latitude Validation:

Criteria: Should be within -90 to 90 degrees.
Example Valid Inputs: 40.712776, -34.052235, 90°
Example Invalid Inputs: -91, 100, xyz

3. Street View Camera Angle Validation:

Criteria: Should be between 0 and 360 degrees.
Example Valid Inputs: 0, 180.5, 360°
Example Invalid Inputs: -10, 400, angle

**Comment by ibrahimBougaoua** [comment]
**Nice idea**

**Comment by shaedrich** [comment]
1\. \+ 2\. Keep in mind, that there could be different notations, so it might not be a good idea to favor one above the other:
- [ISO 6709/D in degrees](https://en.wikipedia.org/wiki/ISO_6709) (e.g. `1.286795, 103.854511`)
- [ISO 6709/F in decimal and CRS](https://en.wikipedia.org/wiki/ISO_6709) (e.g. `50°40'46"N 95°48'26"W 123.45m`)
- `[ lat, lng ]` tuple
- `{ lat: int, lng: int }` object
- etc.
If it's merely about validating numbers, [a custom rule](https://laravel.com/docs/validation#custom-validation-rules) might suffice (also, [`\Illuminate\Validation\Rule`](https://laravel.com/api/10.x/Illuminate/Validation/Rule.html) is [macroable](https://laravel.com/api/10.x/Illuminate/Support/Traits/Macroable.html) 😉)

3\. "Street View Camera Angle" seems oddly specific&mdash;in other words: proprietary validation rules are probably not wanted and it's recommended to create a package for that. However, "angle" might be fine (You can name your field "street_view_camera_angle" but a core validation rule would probably be "angle").

**Reply by asfiaaiman** [reply]
Generally latitude and longitude have these values 1.286795, 103.854511 that are supported by google maps as well

**Reply by shaedrich** [reply]
You didn't read my post, did you? Google Maps is not all there is (goodness gracious, people with their proprietary names these days …). You may not be able to grasp this, but people used geo coordinates way before Google was a thing. Please, please, don't mess up naming just because your horizon is so limited and/or you don't care about use cases others might have. But when that's the case, just create a package, call it "google-maps-stuff-validation" and be done with it.
